tools/pkg/parsers/constraint-parser.toit

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2024 Toitware ApS.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; version
+// 2.1 only.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// The license can be found in the file `LICENSE` in the top level
+// directory of this repository.
+
+import .semantic-version-parser
+
+
+class ConstraintParseResult:
+  prefix/string?
+  semantic-version/SemanticVersionParseResult
+
+  constructor .prefix .semantic-version:
+
+/*
+  A PEG grammer for the constraints
+
+  constraints ::= constraint ( ',' space* constraint )* space*
+  constraint ::= prefix? semantic-version
+  prefix ::= "!=" | '>=' | '<=' | "~>" |  '=' | '>' | '<' | '~' | '^'
+  space := ' ' | '\t'
+  In particular, no '*' productinos as in '1.*', that can be done with '^1.0.0'
+*/
+class ConstraintParser extends SemanticVersionParser:
```

Comment:
I did change this, but in this case it made the code somewhat more convoluted.

Reply:
On second thought, this will not really work well for multiple constraints, so reverted to inheritance. The problem is that having a delegate requires a lot of manipulation of the parser position, where as with inheritance that is taken care of.